### ISSUE TITLE
form keras part recurrent_dropout causes a exploding loses 

### ISSUE THREAD

**Issue by birolkuyumcu** [tensorflow/tensorflow#39676]
recurrent_dropout promotes exploding losses
https://github.com/fchollet/deep-learning-with-python-notebooks/issues/127
https://github.com/fchollet/deep-learning-with-python-notebooks/issues/52

same problem with  Tensorflow :  2.0.0 and  Keras :  **2.2.4-tf**

https://github.com/tensorflow/tensorflow/issues/35724#issuecomment-582712923

when recurrent_dropout is zero there is no problem

**Comment by Saduf2019** [comment]
@birolkuyumcu 
Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?

Make sure you also include the exact command if possible to produce the output included in your test case. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by birolkuyumcu** [comment]
@Saduf2019 
if you check these links there is no only me 
and also known at see this comment
https://github.com/tensorflow/tensorflow/issues/35724#issuecomment-582712923

> "I confirmed numerous times again, that recurrent_dropout promotes exploding gradients (nan loss & weights) "

i use 
same problem with Tensorflow : 2.0.0 and Keras : 2.2.4-tf
with prebuilt tensorflow pip package under windows

**Comment by Reeshad-Khan** [comment]
any solution regarding the issue? I'm using Windows 10 64 bit OS. 
Tensorflow: 1.15.0 and Keras: 2.2.4-tf

**Comment by birolkuyumcu** [comment]
> 
> 
> any solution regarding the issue? I'm using Windows 10 64 bit OS.
> Tensorflow: 1.15.0 and Keras: 2.2.4-tf

dont use  recurrent_dropout for TF 2.0

**Comment by birolkuyumcu** [comment]
i am try wit TF 2.2 at colab 
recurrent_dropout  does not  cause a expanding losses. 
So for TF 2.2 at colab  there is no problem,

**Comment by Reeshad-Khan** [comment]
ohhh... ok mate... thanks... let me check....

**Comment by Reeshad-Khan** [comment]
![rnn](https://user-images.githubusercontent.com/65279631/82457786-ab408580-9ad7-11ea-85f6-c94aff997b25.PNG)
Same again....
I'm using TF 2.2 at colab

**Comment by birolkuyumcu** [comment]
![resim](https://user-images.githubusercontent.com/8981828/82466086-a6280a00-9ac8-11ea-93e0-c99434f0b151.png)

strange !
this my case 
tf 2.2.0 , colab

**Comment by Reeshad-Khan** [comment]
yeah!!! really strange!!!!

**Comment by jvishnuvardhan** [comment]
@birolkuyumcu @Reeshad-Khan Can you please share a standalone code to reproduce the issue? I know `dropout` was an issue in `TF2.0` but not anymore. Thank you.

**Comment by birolkuyumcu** [comment]
https://github.com/fchollet/deep-learning-with-python-notebooks/blob/master/6.3-advanced-usage-of-recurrent-neural-networks.ipynb
cell 28

**Comment by birolkuyumcu** [comment]
Hi 
i write standalone code to reproduce the issue
problem related to GRU and timesteps value
https://gist.github.com/birolkuyumcu/b0ab305dab5a6f1ed3f8c4f59e954a1f

**Comment by jvishnuvardhan** [comment]
@birolkuyumcu I think this was resolved already. Please check the [gist](https://colab.research.google.com/gist/jvishnuvardhan/cbe0be8152adfa6680d1665c0adbba1e/gru_recurrent_dropout_bug.ipynb) with `tf-nightly`. I had checked on local also.

Please verify once and close the issue. If you have any further support type questions, please post them in Stackoverflow as there is a large community to help. GitHub is mainly for tracking bugs. Thanks!

**Comment by CFGrote** [comment]
possibly related?
https://github.com/fchollet/deep-learning-with-python-notebooks/issues/151
